commit 33dfafa90285c0873a24d633877d505ab8e3fc20
Author: Kent Overstreet <kent.overstreet@linux.dev>
Date:   Wed Jun 19 09:55:48 2024 -0400

    bcachefs: Fix safe errors by default
    
    i.e. the start of automatic self healing:
    
    If errors=continue or fix_safe, we now automatically fix simple errors
    without user intervention.
    
    New error action option: fix_safe
    
    This replaces the existing errors=ro option, which gets a new slot, i.e.
    existing errors=ro users now get errors=fix_safe.
    
    This is currently only enabled for a limited set of errors - initially
    just disk accounting; errors we would never not want to fix, and we
    don't want to require user intervention (i.e. to make sure a bug report
    gets filed).
    
    Errors will still be counted in the superblock, so we (developers) will
    still know they've been occuring if a bug report gets filed (as bug
    reports typically include the errors superblock section).
    
    Eventually we'll be enabling this for a much wider set of errors, after
    we've done thorough error injection testing.
    
    Signed-off-by: Kent Overstreet <kent.overstreet@linux.dev>

diff --git a/fs/bcachefs/bcachefs_format.h b/fs/bcachefs/bcachefs_format.h
index 4b98fed1ee9a..e3b1bde489c3 100644
--- a/fs/bcachefs/bcachefs_format.h
+++ b/fs/bcachefs/bcachefs_format.h
@@ -990,8 +990,9 @@ enum bch_version_upgrade_opts {
 
 #define BCH_ERROR_ACTIONS()		\
 	x(continue,		0)	\
-	x(ro,			1)	\
-	x(panic,		2)
+	x(fix_safe,		1)	\
+	x(panic,		2)	\
+	x(ro,			3)
 
 enum bch_error_actions {
 #define x(t, n) BCH_ON_ERROR_##t = n,